source/common/stats/tag_extractor_impl.cc

```diff
@@ -114,13 +114,19 @@ bool TagExtractorStdRegexImpl::extractTag(TagExtractionContext& context, std::ve
                                                      regex_) &&
       match.size() > 1) {
     // remove_subexpr is the first submatch. It represents the portion of the string to be removed.
-    const auto& remove_subexpr = match[1];
+    uint idx;
+    for (idx = 1; idx < match.size(); idx++) {
+      if (match[idx].str() != "") {
```

Comment:
I think this should be instead:

```
-      if (!match[idx].str().empty()) {
+      if (match[idx].matched) {
```

This way we continue to allow empty capturing groups, which would be a regression (maybe add a test for empty group since it appears there is none). Istio does allow setting empty string as tag values, for example.